### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by provinzio** [sabeechen/hassio-google-drive-backup#727]
###### Description:
```
If you have anything else that could help explain what happened, click "Markdown" above and write it here.
```

Just installed the add-on and got this error.

 Addon version: `0.108.4`
 Home Assistant Version: `2022.9.7`
 Supervisor Version: `2022.09.1`
 Supervisor Channel: `stable`
 Hassos Version: `9.0`
 Docker Version: `20.10.17`
 Architecture: `aarch64`
 Machine: `raspberrypi3-64`
 Date: `2022-09-30T10:24:10.509993+00:00`
 Timezone: `Europe/Berlin`
 Failure Time: `2022-09-30T10:21:44.795790+00:00`
 Last Good Sync: `2022-09-30T10:21:43.053254+00:00`
 ###### Exception:
 ```
 
  addon/backup/model/coordinator.py:177 (_sync)
  addon/backup/model/model.py:198 (sync)
  addon/backup/model/model.py:318 (_handleBackupDetails)
  addon/backup/model/backupscheme.py:216 (handleNaming)
  addon/backup/model/backupscheme.py:126 (_buildPartitions)
IndexError: list index out of range

 ```
 Backups:
 ```
 {
    "HomeAssistant": {
        "backups": 0,
        "retained": 0,
        "deletable": 0,
        "name": "HomeAssistant",
        "title": "Home Assistant",
        "latest": null,
        "max": 2,
        "enabled": true,
        "icon": "home-assistant",
        "ignored": 7,
        "detail": "",
        "size": "0.0 B",
        "ignored_size": "10.1 MB",
        "free_space": "50.3 GB"
    },
    "GoogleDrive": {
        "backups": 0,
        "retained": 0,
        "deletable": 0,
        "name": "GoogleDrive",
        "title": "Google Drive",
        "latest": null,
        "max": 6,
        "enabled": true,
        "icon": "google-drive",
        "ignored": 0,
        "detail": "carsten.docktor@gmail.com",
        "size": "0.0 B",
        "ignored_size": "0.0 B",
        "free_space": "5.2 GB"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_BACKUPS_IN_HA": 2,
    "Setting.MAX_BACKUPS_IN_GOOGLE_DRIVE": 6,
    "Setting.DAYS_BETWEEN_BACKUPS": 1.0,
    "Setting.BACKUP_TIME_OF_DAY": "03:00",
    "Setting.BACKUP_PASSWORD": "REDACTED",
    "Setting.GENERATIONAL_DAYS": 3,
    "Setting.GENERATIONAL_WEEKS": 4,
    "Setting.GENERATIONAL_DAY_OF_WEEK": "wed",
    "Setting.GENERATIONAL_DELETE_EARLY": true
}
 ```
 ###### Addon Logs:
 ```
 09-30 12:18:03 INFO [backup.model.coordinator] Syncing Backups
09-30 12:18:03 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
09-30 12:18:03 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/backups
09-30 12:18:03 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:177 (_sync)
  addon/backup/model/model.py:198 (sync)
  addon/backup/model/model.py:318 (_handleBackupDetails)
  addon/backup/model/backupscheme.py:216 (handleNaming)
  addon/backup/model/backupscheme.py:126 (_buildPartitions)
IndexError: list index out of range

09-30 12:18:03 INFO [backup.model.coordinator] I'll try again in 1(ish) minutes
09-30 12:19:23 DEBUG [backup.model.syncer] Sync requested by Coordinator
09-30 12:19:23 INFO [backup.model.coordinator] Syncing Backups
09-30 12:19:23 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
09-30 12:19:23 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/backups
09-30 12:19:23 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:177 (_sync)
  addon/backup/model/model.py:198 (sync)
  addon/backup/model/model.py:318 (_handleBackupDetails)
  addon/backup/model/backupscheme.py:216 (handleNaming)
  addon/backup/model/backupscheme.py:126 (_buildPartitions)
IndexError: list index out of range

09-30 12:19:23 INFO [backup.model.coordinator] I'll try again in 2(ish) minutes
09-30 12:21:05 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/addons/self/options
09-30 12:21:05 INFO [backup.model.coordinator] Syncing Backups
09-30 12:21:05 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
09-30 12:21:05 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/backups
09-30 12:21:05 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:177 (_sync)
  addon/backup/model/model.py:198 (sync)
  addon/backup/model/model.py:318 (_handleBackupDetails)
  addon/backup/model/backupscheme.py:216 (handleNaming)
  addon/backup/model/backupscheme.py:126 (_buildPartitions)
IndexError: list index out of range

09-30 12:21:05 INFO [backup.model.coordinator] I'll try again in 5(ish) minutes
09-30 12:21:42 INFO [backup.drive.drivesource] Saving new Google Drive credentials
09-30 12:21:43 DEBUG [backup.model.syncer] Sync requested by GoogleDrive
09-30 12:21:43 INFO [backup.model.coordinator] Syncing Backups
09-30 12:21:43 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/supervisor/info
09-30 12:21:43 DEBUG [backup.ha.harequests] Making Hassio request: http://supervisor/backups
09-30 12:21:43 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=mimeType%3D%27application%2Fvnd.google-apps.folder%27&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
09-30 12:21:43 INFO [backup.drive.folderfinder] Creating folder "Home Assistant Backups" in "My Drive"
09-30 12:21:43 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?supportsAllDrives=true
09-30 12:21:44 INFO [backup.drive.folderfinder] Saving backup folder: 1t3JX31qlvkw2g1OtF4rD01Dmf-5fmu9s
09-30 12:21:44 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/about?fields=storageQuota%2Cuser
09-30 12:21:44 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%271t3JX31qlvkw2g1OtF4rD01Dmf-5fmu9s%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%2CdriveId%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
09-30 12:21:44 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:177 (_sync)
  addon/backup/model/model.py:198 (sync)
  addon/backup/model/model.py:318 (_handleBackupDetails)
  addon/backup/model/backupscheme.py:216 (handleNaming)
  addon/backup/model/backupscheme.py:126 (_buildPartitions)
IndexError: list index out of range

09-30 12:21:44 INFO [backup.model.coordinator] I'll try again in 10(ish) minutes
 ```
 ###### Supervisor Logs:
 ```
 [32m22-09-30 12:21:05 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-09-30 12:21:05 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m22-09-30 12:21:08 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:08 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:08 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:08 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:13 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:13 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:13 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:43 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m22-09-30 12:21:43 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m22-09-30 12:21:43 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:43 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:21:43 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:22:33 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m22-09-30 12:22:54 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:22:54 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:22:58 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:22:58 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:22:58 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:22:58 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:23:03 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m22-09-30 12:23:05 INFO (MainThread) [supervisor.api.middleware.security] /core/logs access from cebe7a76_hassio_google_drive_backup[0m
[32m22-09-30 12:23:57 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:24:07 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:24:07 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:24:07 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:24:08 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password[0m
[32m22-09-30 12:24:10 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

 ```
 ###### Home Assistant Core Logs:
 ```
 [31m2022-09-30 08:09:50.879 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 08:10:01.645 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 08:10:35.760 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 08:12:19.361 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 08:14:24.603 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 08:17:48.580 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 08:17:58.591 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 08:54:21.444 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 09:29:41.967 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 09:30:35.334 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 09:31:21.105 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 09:51:13.832 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 09:58:15.149 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 10:17:58.820 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 10:29:14.803 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 10:29:34.589 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 10:50:00.502 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:05:31.767 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:06:11.960 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:08:51.518 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:09:13.608 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:12:52.183 ERROR (MainThread) [homeassistant.components.bluetooth.scanner] hci0 (B8:27:EB:DE:A8:B9): Error stopping scanner: [org.bluez.Error.InProgress] Operation already in progress[0m
[31m2022-09-30 11:14:50.138 ERROR (MainThread) [homeassistant.components.bluetooth.scanner] hci0 (B8:27:EB:DE:A8:B9): Error stopping scanner: [org.bluez.Error.InProgress] Operation already in progress[0m
[31m2022-09-30 11:39:58.805 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:42:38.847 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:46:44.645 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:50:17.879 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 11:55:04.181 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m
[31m2022-09-30 12:22:44.290 ERROR (MainThread) [homeassistant.components.alexa.state_report] Error when sending ChangeReport for light.alexa_virtual to Alexa: INVALID_ACCESS_TOKEN_EXCEPTION: Access token is not valid.[0m

 ```

**Comment by provinzio** [comment]
As this error comes from the Generational Backupscheme, I disabled it in the settings, did a backup and activated it again.

Fix should be easy when following the traceback of the exception.

**Comment by sabeechen** [comment]
Been a while since I had a real bug, it happens when all the following are true:
 - Use generational backups
 - Have at least 1 "ignored" backup
 - Have no other backups

I'll release a fix soon, though I won't rush a release since running into this set of conditions should be pretty rare.  This bug has been around for a while and you're the first to report it.

**Comment by provinzio** [comment]
Yea, I guess it doesn't happen that often, that someone sets the config first before trying a first backup. Should be no harm to wait.

And thanks for the fast response and awesome add-on you're sharing. It's a really neat backup solution. 👏

**Comment by sabeechen** [comment]
This should be fixed in the latest verison 0.109.1.  Please re-open if you still run into trouble.
